moirae/estimators/online/utils/model.py

```diff
@@ -57,18 +80,158 @@ def __init__(self,
         self.transients = transients
         self.cell_model = cell_model
         self.asoh = asoh
-        self.input_template = input_template
+        self.inputs = inputs
 
         # Capture the shape of the outputs
-        self._num_output_dimensions = self.cell_model.calculate_terminal_voltage(self.input_template,
-                                                                                 self.transients,
-                                                                                 self.asoh).to_numpy().shape[1]
+        self._num_output_dimensions = self.cell_model.calculate_terminal_voltage(new_inputs=self.inputs,
+                                                                                 transient_state=self.transients,
+                                                                                 asoh=self.asoh).to_numpy().shape[1]
 
     @property
     def num_output_dimensions(self) -> int:
         return self._num_output_dimensions
 
 
+class CellModelWrapper(BaseCellWrapper):
+    """
+    Base link between the :class:`~moirae.model.base.CellModel` and the numpy-only interface of
+    :class:`~moirae.estimators.online.filters.base.BaseFilter` filter implementations. This particular wrapper does not
+    touch the A-SOH, but only uses it for predictions.
+    """
+    def __init__(self,
+                 cell_model: CellModel,
+                 asoh: HealthVariable,
+                 transients: GeneralContainer,
+                 inputs: InputQuantities) -> None:
+        super().__init__(cell_model=cell_model, asoh=asoh, transients=transients, inputs=inputs)
+
+        self.num_transients = transients.to_numpy().shape[1]
+
+    @property
+    def num_hidden_dimensions(self) -> int:
+        return self.num_transients
+
+    def update_hidden_states(self,
+                             hidden_states: np.ndarray,
+                             previous_controls: np.ndarray,
+                             new_controls: np.ndarray) -> np.ndarray:
+        """

```

Comment:
Omit the docstring here. Autodocumentation tools will retrieve the definition from the base class, so we don't want two nearly-identical copies.

Reply:
Done!